src/main/java/org/simlar/simlarserver/services/pushnotification/ApplePushNotification.java

```diff
@@ -171,6 +171,13 @@ String requestVoipPushNotification(final String url, final String urlPin, final
             log.warn("received unexpected response status '{}'", statusCode);
         }
 
-        return StringUtils.join(response.getHeaders().get("apns-id"), ", ");
+        final String apnsId = StringUtils.join(response.getHeaders().get("apns-id"), ", ");
+        if (StringUtils.isEmpty(apnsId)) {
```

Comment:
Concatenating null or empty apns-ids headers will yield to an non-empty `apnsId`. I hate this behavior of `join`, too.